stdlib/messaging/nats/src/main/java/org/ballerinalang/nats/streaming/consumer/Attach.java

```diff
@@ -41,8 +42,15 @@ public static void streamingAttach(ObjectValue streamingListener, ObjectValue se
         ConcurrentHashMap<ObjectValue, StreamingListener> serviceListenerMap =
                 (ConcurrentHashMap<ObjectValue, StreamingListener>) streamingListener
                         .getNativeData(STREAMING_DISPATCHER_LIST);
-        serviceListenerMap.put(service, new StreamingListener(service, BRuntime.getCurrentRuntime(),
-                                                              streamingListener.getObjectValue("connection")
-                                                                      .getStringValue(Constants.URL)));
+        boolean manualAck = getAckMode(service);
+        serviceListenerMap.put(service, new StreamingListener(service, manualAck, BRuntime.getCurrentRuntime(),
+                streamingListener.getObjectValue("connection")
+                        .getStringValue(Constants.URL)));
```

Comment:
Shall we combine this line?

Reply:
https://github.com/ballerina-platform/ballerina-lang/pull/20658#discussion_r367913690